### PR TITLE
Create unit tests for fullness

### PR DESCRIPTION
**Pull request by aahsue**
## Description

Created unit tests using the Mock Distance Sensor and the Fullness Class. Checked if mock distance sensor can be constructed, that it returns the correct distance, that it can loop through the distances, and that the Fullness class returns the correct fullness and if it can recognize if a distance is valid.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code improvement (a non-breaking change that improves the quality of the code)
- [ ] Documentation Change (update to the documentation with no changes to the code)

## Testing and Verification Instructions

1. Build and compile and there are no errors
2. Run the unit tests and they all pass

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: test/test_desktop_fullness/testFullness.cpp [hunk 1]
```diff
@@ -1,17 +1,65 @@
 #include "FullnessMetric.hpp"
+#include "MockDistance.hpp"
 #include <unity.h>
+#include <vector>
 
-// TODO: Add more tests
+void test_can_construct_mock_distance_sensor()
+{
+    Fullness::MockDistance mockDistanceSensor({3, -6, 27, 114, 9});
+}
```
Comment (patrickanguyen) [inline test/test_desktop_fullness/testFullness.cpp:9]: remove

File: test/test_desktop_fullness/testFullness.cpp [hunk 2]
```diff
@@ -1,17 +1,65 @@
...
+        {
+            tempCounter++;
+        }
+    }
+}
+
+void test_returns_correct_fullness()
+{
+    uint32_t binHeight = 10;
+    int32_t calculatedAverageDistance = 5; // should be 5.5
```
Comment (patrickanguyen) [inline test/test_desktop_fullness/testFullness.cpp:43]: this should be a float

File: test/test_desktop_fullness/testFullness.cpp [hunk 3]
```diff
@@ -1,17 +1,65 @@
...
+    std::vector<int32_t> distances = {7, 8, 3, 2, 10, 4, 20, 0};
+    Fullness::MockDistance mockDistanceSensor(distances);
+
+    Fullness::FullnessMetric mockFullness(binHeight, mockDistanceSensor);
+    TEST_ASSERT_EQUAL_INT(calculatedAverageDistance, (mockFullness.getFullness()));
```
Comment (patrickanguyen) [inline test/test_desktop_fullness/testFullness.cpp:48]: getFullness() returns a float

File: test/test_desktop_fullness/testFullness.cpp (hunk 2 above)
Comment (patrickanguyen) [inline test/test_desktop_fullness/testFullness.cpp:43]: Make these const

**Review by patrickanguyen — CHANGES_REQUESTED** [review]
More meaningful tests needed to be added in the future, but we can just merge this once these changes are made

**Comment by limenilbuz** [comment]
lgtm

**Review by sagarbpatel31 — COMMENTED** [review]
Lgtm now

**Review by patrickanguyen — APPROVED** [review]

**Pull request merged**